into/backends/iopro.py

```diff
@@ -0,0 +1,159 @@
+from __future__ import absolute_import, print_function
+
+import sqlalchemy as sa
+import numpy as np
+from toolz import memoize
+
+from ..convert import convert
+#This import will fail if iopro is not installed.
+#The goal is to simply have all iopro converters
+#  be missing if iopro isn't around
+import iopro
+import iopro.pyodbc
+
+@memoize
+def get_iopro_db_engine(engine):
+    """Return an iopro.pyodbc-ized sqlachemy engine
+
+    Parameters
+    ----------
+    engine - A sqlalchemy engine
+
+    Return
+    ------
+    An new sqlalchemy engine that will use iopro.pyodbc to make the
+    connection to the db. If the connection cannot be made, a
+    NotImplementedError exception will be raised so that a calling
+    "convert()" or "into()" will be able to use a different
+    sql->ndarray conversion path.
+    
+    Notes
+    -----
+    This function will try to make a db connection
+      that iopro.pyodbc can use from an existing
+      sqlalchemy engine/connection.
+    Example connection string for a mysql connection:
+    connectionstring = "DRIVER={mysql};\
+                       SERVER=localhost;\
+                       DATABASE=continuum_db;\
+                       UID=continuum_user;\
+                       PASSWORD=continuum_test;\
+                       OPTION=3;"
+
+    The DRIVER={mysql} part comes from the odbcinst.ini file that *MUST*
+      exist (on *nix) in order for ODBC to be able to find the correct
+      driver.
+    Furthermore, this file should have sections that *MATCH* the
+      sqlalchemy driver names.
+    i.e. if sqlalchemy.engine.get_backend_name() returns "mysql",
+      then your odbcinst.ini file should have a section resembling this:
+      [mysql]
+      Description= MySQL driver
+      Driver= /usr/lib/x86_64-linux-gnu/odbc/libmyodbc.so
+
+    An odbc.ini file is NOT required by this function.
+    My understanding is that odbc.ini is only required for pre-defined
+      data-source names (DSNs).
+    No DSNs are used here because we extract all necessary data from
+      the sqlalchemy engine.
+    """
+    url = engine.url
+    connect_args = url.translate_connect_args()
+
+    server = connect_args.get('host','')
+    uid = connect_args.get('username','')
+    password = connect_args.get('password','')
+    database = connect_args.get('database','')
+    port = connect_args.get('port','')
+    option = connect_args.get('option','')
+
+    driver = dialect = url.get_backend_name()
+    
+    if driver == "mysql":
+        #option=3 is required by mysql
+        option = "3"
+        dialect = "mysql+pyodbc"
+    elif driver == "mssql":
+        dialect = "mssql+pyodbc"
+    else:
+        raise NotImplementedError("sqlalchemy does not support the {} database "\
+                                  "with the pyodbc dialect yet.".format(driver))
```

Comment:
I might not have understood your comments earlier about postgres, but is it not possible to use Postgres here?  Presumably we have information about where the database is and the authentication we need.  

Also, any love for `sqlite`?